hw_toledo_scale/controllers/main.py

```diff
@@ -0,0 +1,557 @@
+                _logger.debug(
+                    '[CALCULATE CHECKSUMS] Checksum and correction value generation with the number {}'.format(z))
+
+                # choose a random checksum (cs)
+                cs = self._generate_random_hex_code()
+                cs_size = len(cs) * 4
+                cs_bin = bin(int(cs, 16))[2:].zfill(cs_size)
+
+                # apply first bit of z on checksum
+                cs_bin_encoded = self._rotate_left(cs_bin, int(z[0], 16))
+                cs_hex_ascii_encoded = "{:04X}".format(int(cs_bin_encoded, 2))
+
+                # generate kw
+                kw = self._get_kw_value(toledo_polynomial_p, int(cs, 16))
+                kw_size = len(kw) * 4
+                kw_bin = bin(int(kw, 16))[2:].zfill(kw_size)
+
+                # apply second bit of z on kw
+                kw_bin_encoded = self._rotate_right(kw_bin, int(z[1], 16))
+                kw_hex_ascii_encoded = '{:04X}'.format(int(kw_bin_encoded, 2), 'X')
+
+                _logger.debug('[CALCULATE CHECKSUMS] Checksum chosen before encoding: {}'.format(cs))
+                _logger.debug('[CALCULATE CHECKSUMS] Checksum after encoding: {}'.format(cs_hex_ascii_encoded))
+                _logger.debug(
+                    '[CALCULATE CHECKSUMS] Correction value after encoding: {}'.format(kw_hex_ascii_encoded))
+            except Exception as e:
+                self.set_status(
+                    'error',
+                    "During generation of checksum and correction value with Exception {}".format(e))
+                self.device = None
+        return cs_hex_ascii_encoded, kw_hex_ascii_encoded
+
+    def send_checksums(self, cs, kw, device):
+        connected, error = False, None
+        try:
+            self.send_record_10(cs, kw, device)
+            answer_checksum = self._get_raw_response(device)
+            if answer_checksum == NAK:
+                _logger.debug('[SEND CHECKSUM] Frame received is NAK frame')
+                self.set_status(
+                    'error',
+                    'Could not connect on scale {} with protocol {}. Wrong or Nak response. Answer: {}'.format(
+                        self.path_to_scale, self.protocol.name, answer_checksum))
+                error = self.request_status_information(device)
+            elif answer_checksum == ACK:
+                _logger.debug(
+                    '[SEND CHECKSUM] Frame received is ACK frame. POS transmits EOT ENQ')
+                self.send_eot_enq(device)
+                answer_status = self._get_raw_response(device)
+                if answer_status == NAK:
+                    error = self.request_status_information(device)
+                else:
+                    record_no, error = self._parse_scale_answer(
+                        self.protocol, answer_status, self.protocol.status_regexp, self.protocol.parse_record_no)
+
+                    if record_no == u'11':
```

Comment:
```suggestion
                    if record_no == '11':
```

Inutile de mettre le `u` en python3